### PR TITLE
Tempo: TraceQL table and editor

### LINKED ISSUE #54997: Tempo - Collapsible trace rows in TraceQL table
Sync with the Grafana BI team (who own the [Table component](https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/Table/Table.tsx)) to add a way to configure [rows to be collapsible](https://react-table-v7.tanstack.com/docs/examples/expanding) or allow [expandable sub-components](https://react-table-v7.tanstack.com/docs/examples/sub-components).

Next, update the TraceQL response transformer to generate a table with a small set of spans inside the collapsable trace row. Regularly sync with the Tempo team and reach out to the UX team to find the best possible solution.

### PR DESCRIPTION
**Pull request by adrapereira**
This PR updates the response table of the Tempo datasource TraceQL tab, by integrating it with the sub-tables changes from #58682. The list of spans for each trace is aggregated in a dataframe which is then fed into the Table component in the `subData` field. The dynamic attributes sent by Tempo are used to create fields in each frame, so the sub-table for each trace will have its own set of columns, which is handy for complex queries with a lot of attributes, since not all attributes may match for all traces.

It also includes updates to the query editor, for example, multi-line query support or better handling of boolean keywords.

https://user-images.githubusercontent.com/4313971/203817161-9929bf2c-2ed9-47c0-8735-c884f49bc57a.mov

Fixes https://github.com/grafana/grafana/issues/54997

### CODE REVIEW

File: public/app/plugins/datasource/tempo/traceql/TraceQLEditor.tsx [hunk 1]
```diff
@@ -92,19 +92,30 @@ function setupActions(editor: monacoTypes.editor.IStandaloneCodeEditor, monaco:
...
-
     contextMenuOrder: 1.5,
-
     run: function () {
       onRunQuery();
     },
   });
 }
 
+function setupAutoHeight(editor: monacoTypes.editor.IStandaloneCodeEditor) {
```
Comment (joey-grafana) [inline public/app/plugins/datasource/tempo/traceql/TraceQLEditor.tsx:104]: nit: should call `setupAutoSize` as setting both width and height

**Review by joey-grafana — APPROVED** [review]
Overall LGTM 👍 

Suggestions (non-blocking)
- navigating from TraceID -> TraceQL preserves the trace id in the main search field. UX could be improved here, perhaps by simply removing the trace id when navigating to TraceQL (unless there's a specific reason we preserve the fields input).
- column widths could be aligned a bit better in the sub table. In the video the span id and ip headers have different widths (looks like width evenly distributed). Span id could prob have a fixed width for better UX.
- clicking span id in top row in sub table does not work as id mismatch
https://user-images.githubusercontent.com/90795735/204275752-fb11dd27-7381-4a83-ba05-b7f51b0d5444.mov
- no data needs some design update
Screenshot 2022-11-28 at 11 48 57

**Review by aocenas — APPROVED** [review]
Also looks good to me don't have much to add on top of @joey-grafana comments.

**Comment by aocenas** [comment]
Maybe just in context of:
> navigating from TraceID -> TraceQL preserves the trace id in the main search field. UX could be improved here, perhaps by simply removing the trace id when navigating to TraceQL (unless there's a specific reason we preserve the fields input).

I know we have a separate task for the UI of query editor, but it seems like a really non controversial low hanging fruit to just join the TraceID and TraceQL tab into one. So that if somebody turns on the feature toggle it won't add a new tab just changes the TraceID -> TraceQL. To know what query to do it should be easy to just regex match the query to see if it is a traceID or not.

**Pull request merged**